core/services/pg/connection.go

```diff
@@ -50,5 +50,38 @@ func NewConnection(uri string, dialect dialects.DialectName, config ConnectionCo
 	db.SetMaxOpenConns(config.ORMMaxOpenConns())
 	db.SetMaxIdleConns(config.ORMMaxIdleConns())
 
+	err = disallowReplica(db)
+	if err != nil {
+		return nil, err
+	}
 	return db, nil
```

Comment:
Looks like you could simply these 5 lines to just 1:
```
return disallowReplica(db)
```
If the caller detects `err != nil` then it can't make any assumptions about what db is anyway.